samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -139,7 +139,12 @@ class KafkaSystemAdmin(
    * Replication factor for the Changelog topic in kafka
    * Kafka properties to be used during the Changelog topic creation
    */
-  topicMetaInformation: Map[String, ChangelogInfo] = Map[String, ChangelogInfo]()) extends ExtendedSystemAdmin with Logging {
+  topicMetaInformation: Map[String, ChangelogInfo] = Map[String, ChangelogInfo](),
+
+  /**
+   * Kafka properties to be used during the intermediate topic creation
+   */
+  intermediateStreamProperties: Map[String, Properties] = Map()) extends ExtendedSystemAdmin with Logging {
```

Comment:
s/intermediateStreamProperties/streamProperties
Can also consolidate this with other streams like coordinator and use a single map perhaps?

Reply:
I think this is beyond the scope of this pr. The refactoring should be separate patch and needs more testing than what I have it here.

Reply:
In that case do we really need this extra stateful parameter at all? Can't we simply use the properties from the `StreamSpec` class?

The `createStream(StreamSpec)` can internally construct the right `KafkaStreamSpec` by parsing the properties from the passed in `StreamSpec` instance ?

Reply:
Passing in specific kafka properties in StreamSpec and then construct KafkaStreamSpec from it seems not a good idea. As far as I understand, StreamSpec should be general and can be materialized into different physical stream.